### DISCUSSION TITLE
Compile only the active workspace instead of all

### DISCUSSION THREAD

**Discussion by jerehut** [glenn2223/vscode-live-sass-compiler#185] [Ideas]
How can I make the "Watch Sass" button to only generate CSS for the active workspace? It just generated CSS files for 10 different workspaces for me. Did I do something wrong?

**Comment by glenn2223** [comment]
> Did I do something wrong?

No, nothing wrong. The extension is designed to watch the entire solution, not individual workspaces.

----

The closest you can get is to disable [`liveSassCompile.settings.compileOnWatch`](https://github.com/glenn2223/vscode-live-sass-compiler/blob/master/docs/settings.md#livesasscompilesettingscompileonwatch). Then only saving a partial file compiles the entire project

**Comment by glenn2223** [comment]
Thinking about this. I could possibly add a setting that works on the active workspace only.

It's worth noting that workspaces are all treaded equally[<sup>[1]</sup>](https://code.visualstudio.com/api/references/vscode-api#WorkspaceFolder), so there is no such thing as an "active" or "primary" workspace. However, if there is an active file open then I can determine its workspace folder and work to that.

### Feature
This means that saving a partial file will only compile files in the current workspace

**Comment by glenn2223** [comment]
## Discussion point

### Name and Setup

Name: `liveSassCompile.settings.workspacesAreLinked`  
 
Setup: A `boolean`, defaulted to true. Set to `false` to work on only the current workspace.

**Comment by glenn2223** [comment]
## Discussion point

### What to do when...

When `false` and there is no active file, do we:
  - throw an error and stop processing, or
  - instead, compile everything  
 
Personally, I prefer the latter

**Reply by glenn2223** [reply]
Maybe supporting a bool and object. Just false will compile everything. An object will support choosing how to handle no active files

**Comment by Team-Startup** [comment]
In my case, this option is important. 
We would like to use the ‘live-sass-compiler’ module, but without liveSassCompile.settings.workspacesAreLinked, this is not possible. We have about twenty independent root folders, and compilation must not be performed on all of them because it is too heavy, and some folders must not be recompiled.

The solution with workspacesAreLinked is exactly what we want 👍 . 

Just in case there is no file selected, I think we should do nothing, except perhaps generate an error (I did not fully understand your solution).

**Reply by Team-Startup** [reply]
Hi @glenn2223 ,
Any news on this development?
How can I help? Maybe I should contribute financially to help?
Have a good day.
Vincent

**Reply by glenn2223** [reply]
Sorry Vincent, not made any progress yet.

Work, uni studies, family life, etc. has meant I have little time to spend on my open source stuff - plus the burnout 😮‍💨

A financial contribution would definitely go a long way, currently there's 2.5m installs and I'm getting ... well, nothing (not helping the burnout). No pressure, but, yeah... a financial contribution, of some capacity, would go a long way - you can see all the donation methods in the [`README.md`](https://github.com/glenn2223/vscode-live-sass-compiler/blob/master/README.md) I finally got around to updating

**Comment by glenn2223** [comment]
I hope you do manage to get around to financially supporting the project; but with the work I've been doing recently for v7 - I didn't want to essentially "hold this hostage" because of the promise of money. To which, I've worked on this and I am about to push to main.

Again, I hope you get around to some small financial contribution. If your business has a marketing budget, sponsors technically get some marketing when they support through GitHub 😁 

---

You can wait for a pre-release (coming soon), or package it up yourself to get a head start (just follow the publish.yaml instructions, replacing `vsce publish` with `vsce package`)
